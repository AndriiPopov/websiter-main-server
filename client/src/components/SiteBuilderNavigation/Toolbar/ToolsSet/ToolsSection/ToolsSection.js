import React from 'react';
import PropTypes from 'prop-types';

import classes from './ToolsSection.module.css';

export const ToolsSection = props => {
    return (
        <td className={classes.Section}>
            <div className={classes.ContentDiv}>
                <table className={classes.ContentTable}>
                    <tbody>
                        <tr>
                            {props.children}
                        </tr>
                    </tbody>
                </table>
            </div>
            <div className={classes.TitleDiv}>
                {props.label}
            </div>
        </td>
    );
};

ToolsSection.propTypes = {
    label: PropTypes.string
};

export default ToolsSection;