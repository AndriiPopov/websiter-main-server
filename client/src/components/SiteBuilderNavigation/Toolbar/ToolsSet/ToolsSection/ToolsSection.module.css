.TitleDiv {
    text-align: center;
}

.ContentDiv {
    height: 74px;
}

.ContentTable {
    height: 100%;
    border-collapse: collapse;
}

.Section {
    height: 100px;
    padding: 0px !important;
    padding-left: 3px !important;
    padding-right: 3px !important;
    border-right: 1px dashed #222;
}