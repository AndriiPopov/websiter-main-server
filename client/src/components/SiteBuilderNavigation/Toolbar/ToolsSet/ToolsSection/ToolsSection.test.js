import React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { toolsSection as ToolsSection } from './ToolsSection';

configure({adapter: new Adapter()});

describe('ToolsSection', () => {
    it('should be defined', () => {
        expect(ToolsSection).toBeDefined();
    });

    it('it should render a td with children and title', () => {
        const tree = shallow(
            <ToolsSection
                label="Test section" >
                <p>Test</p>
            </ToolsSection>
        );
        expect(toJson(tree)).toMatchSnapshot();
    });
});