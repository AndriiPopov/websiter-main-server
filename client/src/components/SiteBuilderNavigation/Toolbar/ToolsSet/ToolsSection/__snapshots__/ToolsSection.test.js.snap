// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolsSection it should render a td with children and title 1`] = `
<td
  className="Section"
>
  <div
    className="ContentDiv"
  >
    <table
      className="ContentTable"
    >
      <tbody>
        <tr>
          <p>
            Test
          </p>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="TitleDiv"
  >
    Test section
  </div>
</td>
`;
